### DISCUSSION TITLE
CFAST syntax error

### DISCUSSION THREAD

**Discussion by CarloPollo** [firemodels/cfast#2109] [Q&A]
Good morning, 

I make a premise, I am usually a user of FDS and have never used CFAST. 
This time for a “small project” of a very simple geometry logistics I wanted to try to use CFAST.

I have created the geometry of my building, which is a rectangle with dimensions of 37.20 m x 70.40 m x 14 m(h), I have inserted a 49 MW ultra fast HRR curve (peak HRR at 525 s) and I have inserted an opening at the back of the building. 

Inserting a second compartment on the roof of the main building gives me this error: 

“_-------------------- Input File Syntax Check 7 (Wearhouse).
Fatal: Visualization 1 width position is less than 0 m or greater than compartment width.
Fatal: Visualization 0 width position is less than 0 m or greater than compartment height_.”

I checked via smokeview and the geometry is positioned correctly, what could this error be caused by? 

I tried looking in the user's guide but couldn't find anything. 

I attach the CFAST file
![CFAST-Message](https://github.com/firemodels/cfast/assets/164619172/5dd7fcf4-d4bb-42e0-9df9-674aab1633ab)
[Input-file.pdf](https://github.com/firemodels/cfast/files/15370232/Input-file.pdf)

**Comment by rpeacoc** [comment]
These messages are from the visualization inputs for a CFAST slice file. Unlike FDS, the inputs got CFAST are relative to the compartment origin, not the overall structure origin. If you want to upload the input file, we can take a look.RichardSent from my iPadOn May 19, 2024, at 1:38 PM, CarloPollo ***@***.***> wrote:﻿
Good morning,
I make a premise, I am usually a user of FDS and have never used CFAST.
This time for a “small project” of a very simple geometry logistics I wanted to try to use CFAST.
I have created the geometry of my building, which is a rectangle with dimensions of 37.20 m x 70.40 m x 14 m(h), I have inserted a 49 MW ultra fast HRR curve (peak HRR at 525 s) and I have inserted an opening at the back of the building.
Inserting a second compartment on the roof of the main building gives me this error:
“-------------------- Input File Syntax Check 7 (Wearhouse).
Fatal: Visualization 1 width position is less than 0 m or greater than compartment width.
Fatal: Visualization 0 width position is less than 0 m or greater than compartment height.”
I checked via smokeview and the geometry is positioned correctly, what could this error be caused by?
I tried looking in the user's guide but couldn't find anything.
I attach the CFAST file
CFAST-Message.PNG (view on web)
Input-file.pdf

**Reply by CarloPollo** [reply]
thanks dr. Peacock 

The “error” message only occurs when I insert the second compartment, I checked and it would appear that the slices are correctly inserted with respect to the main compartment. 

When I entered the outputs, I entered the slices related to all the compartments, and so with respect to compartment 2 they would be out. By changing the slice output compartment and entering only compartment 1 the message no longer appears. 

Thank you!

I don't know if I should open another trhead, but I have a second doubt. 
If I have a raised volume with a vertical opening, what do I do to connect the two volumes and use the window for ventilation as in the attached file?

[Warehouse.zip](https://github.com/firemodels/cfast/files/15370580/Warehouse.zip)
